h/services/user_signup.py

```diff
@@ -59,7 +63,14 @@ def signup(self, require_activation=True, **kwargs):
         # user's password.
         password = kwargs.pop('password', None)
 
+        # Extract any passed identities for this new user
+        identities = kwargs.pop('identities', [])
+
         user = User(**kwargs)
+
+        # Add identity relations to this new user, if provided
+        user.identities = [UserIdentity(user=user, **i_args) for i_args in identities]
```

Comment:
I _assume_ that if this raises, the `user` created on line 59 won't get persisted in the db (the transaction will fail/rollback). If this is not a safe assumption, please alert me!

Reply:
I think that's safe. The entire request is wrapped in a transaction and if anything raises at any point, the whole transaction is rolled back